src/java.base/share/classes/java/io/Reader.java

```diff
@@ -207,6 +208,10 @@ public int read(char[] cbuf, int off, int len) throws IOException {
                 return n;
             }
 
+            public String readAllChars() throws IOException {
+                return cs.toString();
```

Comment:
Should this be returning a substring of `cs` starting  at `next`?

Reply:
You are correct on both counts. That was pretty lame on my part. Will update. Thanks.

Reply:
Fixed in dc84bd1 and updated test to verify `cs` starting at `next`. Thanks @j3graham for catching it!